packages/mui-material/src/Modal/Modal.js

```diff
@@ -152,16 +153,23 @@ const Modal = React.forwardRef(function Modal(inProps, ref) {
     childProps.onExited = onExited;
   }
 
-  const RootSlot = slots?.root ?? components.Root ?? ModalRoot;
-  const BackdropSlot = slots?.backdrop ?? components.Backdrop ?? BackdropComponent;
-
   const rootSlotProps = slotProps?.root ?? componentsProps.root;
-  const backdropSlotProps = slotProps?.backdrop ?? componentsProps.backdrop;
+  const backdropSlotProps = slotProps.backdrop ?? componentsProps?.backdrop;
+
+  const externalForwardedProps = {
+    slots: {
+      root: slots.root ?? components.Root,
+      backdrop: slots.backdrop ?? components.Backdrop,
+    },
+    slotProps: {
+      root: slotProps.root ?? componentsProps.root,
+      backdrop: backdropSlotProps,
+    },
+  };
```

Comment:
Could this be?:

```suggestion
  const externalForwardedProps = {
    slots: {
      root: components.Root,
      backdrop: components.Backdrop,
      ...slots,
    },
    slotProps: {
      ...componentsProps,
      ...slots
    },
  };
```

This would allow us to remove the `rootSlotProps` and `backdropSlotProps`variables.

Reply:
@DiegoAndai refactored in this commit https://github.com/mui/material-ui/pull/42150/commits/f7a232910c3c74f099412ebdd86bf7927a784fa0, PR is ready for next review